src/languages/haxe.js

```diff
@@ -47,11 +50,20 @@ export default function(hljs) {
       hljs.QUOTE_STRING_MODE,
       hljs.C_LINE_COMMENT_MODE,
       hljs.C_BLOCK_COMMENT_MODE,
-      hljs.C_NUMBER_MODE,
+      {
+        className: 'number',
+        begin: HAXE_NUMBER_RE,
+        relevance: 0
+      },
+      {
+        className: 'built_in',
+        begin: "\\$" + IDENT_RE,
```

Comment:
What kind of identifier is this exactly?

Reply:
The `$` is used to inject a variable into a macro template.

Without going into too much detail, the best way I can describe it is it's for a system similar to JavaScript template strings. Injects a variable into the template:
```js
`the value of xyz is: $xyz`
```

But instead of strings, it's for generating expressions:
```haxe
var fieldName = "bar";

// generate programmatic structure representation of "foo.bar" expression
var expressionData = macro foo.$fieldName;
```

As you can see, Github (and other Haxe grammars) give it a distinct yellow? So that's what I was going for, but sorry if I used wrong className.

Reply:
Looks like a variable, no?  I think if it's a variable we'd want to use `variable`.